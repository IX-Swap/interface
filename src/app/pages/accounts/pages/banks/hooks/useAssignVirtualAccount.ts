import { getIdFromObj } from 'helpers/strings'
import { ValidCurrency } from 'helpers/types'
import { useAuth } from 'hooks/auth/useAuth'
import { useServices } from 'hooks/useServices'
import { useMutation, useQueryCache } from 'react-query'
import { virtualAccounts } from 'config/apiURL'
import { virtualAccountQueryKeys } from 'config/queryKeys'

export const useAssignVirtualAccount = (
  callback?: () => void,
  isAdditional: boolean | undefined = false
) => {
  const queryCache = useQueryCache()
  const { apiService, snackbarService } = useServices()
  const { user } = useAuth()
  const authUserId = getIdFromObj(user)
  const assignVirtualAccount = async (args: {
    currency: ValidCurrency
    userId?: string
  }) => {
    return await apiService.post(virtualAccounts.assign, {
      currency: args.currency,
      userId: args.userId ?? authUserId
    })
  }
  return useMutation(assignVirtualAccount, {
    onSuccess: async () => {
      callback?.()
      void snackbarService.showSnackbar(
        isAdditional
          ? 'Request to add app account sent'
          : 'Account has been assigned successfully!',
        'success'
      )
      await queryCache.invalidateQueries(virtualAccountQueryKeys.getByUserId)
    },
    onError: async (error: any) => {
      void snackbarService.showSnackbar(error.message, 'error')
      callback?.()
    }
  })
}
