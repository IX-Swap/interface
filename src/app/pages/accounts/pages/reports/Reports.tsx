import React from 'react'
import { Box, Grid, Typography } from '@mui/material'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { Filters } from './components/Filters/Filters'
import { ReportsTable } from 'app/pages/accounts/pages/reports/components/ReportsTable/ReportsTable'

export const Reports: React.FC = () => {
  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Reports' />
      </Grid>
      <Grid item>
        <VSpacer size={'small'} />
        <Typography variant={'h4'}>Statements</Typography>
        <VSpacer size={'small'} />
      </Grid>
      <Grid
        item
        container
        justifyContent={'space-between'}
        alignItems={'center'}
      >
        <Grid item>
          <Box marginRight={2} marginTop={2} marginBottom={2}>
            <Typography variant={'body1'}>
              View, manage and track your accounting reports and taxations.
            </Typography>
          </Box>
        </Grid>
        <Grid item>
          <Filters />
        </Grid>
      </Grid>
      <Grid item>
        <VSpacer size={'medium'} />
        <ReportsTable />
      </Grid>
    </Grid>
  )
}
