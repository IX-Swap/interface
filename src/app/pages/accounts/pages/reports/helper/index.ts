import { formatAmount } from 'helpers/numbers'

export const formatValue = (value: string | number) =>
  typeof value === 'number' ? formatAmount(value) : value
