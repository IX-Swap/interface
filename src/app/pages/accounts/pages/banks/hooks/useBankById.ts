import { UseQueryData } from 'hooks/useParsedData'
import { useServices } from 'hooks/useServices'
import { useQuery } from 'react-query'
import { useAuth } from 'hooks/auth/useAuth'
import { Bank } from 'types/bank'
import { getIdFromObj, isEmptyString } from 'helpers/strings'
import { banksQueryKeys } from 'config/queryKeys'
import { accountsURL } from 'config/apiURL'

export interface UseBankByIdArgs {
  bankId?: string
  ownerId?: string
}

export const useBankById = (args: UseBankByIdArgs): UseQueryData<Bank> => {
  const { bankId, ownerId } = args
  const { apiService } = useServices()
  const { user } = useAuth()
  const userId = ownerId ?? getIdFromObj(user)
  const uri = accountsURL.banks.getById(userId, bankId)

  const getBank = async () => await apiService.get<Bank>(uri)
  const { data, ...rest } = useQuery(
    [banksQueryKeys.getById, userId, bankId],
    getBank,
    { enabled: !isEmptyString(bankId) && !isEmptyString(userId) }
  )

  return {
    ...rest,
    data: data?.data
  }
}
