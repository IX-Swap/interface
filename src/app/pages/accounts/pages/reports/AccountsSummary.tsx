import React from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { Actions } from './components/Actions/Actions'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useGetActivitySummary } from 'app/pages/accounts/hooks/useGetActivitySummary'

export const AccountsSummary: React.FC = () => {
  const { data, isLoading } = useGetActivitySummary()
  console.log(data, isLoading)

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Activity Summary' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo>
        <Actions onContractClick={() => null} onExpandClick={() => null} />
      </ReportsInfo>

      <Grid item>
        <VSpacer size={'medium'} />

        <ReportsAccordion summary={'Open Positions'}>
          <div>Open Positions</div>
        </ReportsAccordion>

        <ReportsAccordion summary={'Cash Report'}>
          <div>Cash Report</div>
        </ReportsAccordion>
      </Grid>
    </Grid>
  )
}
