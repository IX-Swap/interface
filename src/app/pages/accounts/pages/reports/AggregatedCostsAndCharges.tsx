import React, { useEffect } from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useFeeAndCharges } from 'app/pages/accounts/hooks/useFeeAndCharges'
import { FeesTable } from 'app/pages/accounts/pages/reports/components/FeesTable/FeesTable'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { NoData } from 'app/pages/accounts/pages/reports/components/NoData/NoData'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'

export const AggregatedCostsAndCharges: React.FC = () => {
  const { data, isLoading } = useFeeAndCharges()
  const { updateFilter } = useQueryFilter()

  useEffect(() => {
    updateFilter('expandedSections', 'Fees (SGD),Fees (USD)')
  }, [])

  const hasSGDData = !!data?.sgd?.length
  const hasUSDData = !!data?.usd?.length

  const renderContent = () => {
    if (!hasSGDData && !hasUSDData) {
      return <NoData />
    }
    if (hasSGDData) {
      return (
        <ReportsAccordion summary={'Fees (SGD)'}>
          <FeesTable accounts={data.sgd} total={data.totalSgd} />
        </ReportsAccordion>
      )
    }

    if (hasUSDData) {
      return (
        <ReportsAccordion summary={'Fees (USD)'}>
          <FeesTable accounts={data.usd} total={data.totalUsd} />
        </ReportsAccordion>
      )
    }
  }

  if (isLoading) {
    return <LoadingIndicator />
  }

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Aggregated Costs and Charges' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />
        {renderContent()}
      </Grid>
    </Grid>
  )
}
