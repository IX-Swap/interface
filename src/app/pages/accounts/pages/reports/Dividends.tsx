import React, { useEffect } from 'react'
import { Grid } from '@mui/material'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useDividends } from 'app/pages/accounts/hooks/useDividends'
import { DividendsTable } from 'app/pages/accounts/pages/reports/components/DividendsTable/DividendsTable'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { NoData } from 'app/pages/accounts/pages/reports/components/NoData/NoData'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'

export const Dividends: React.FC = () => {
  const { updateFilter } = useQueryFilter()
  const { data, isLoading } = useDividends()

  useEffect(() => {
    updateFilter('expandedSections', 'Dividends')
  }, [])

  const renderContent = () => {
    if (data === undefined || data.length < 1) {
      return <NoData />
    }
    return (
      <ReportsAccordion summary={'Dividends'}>
        <DividendsTable data={data} />
      </ReportsAccordion>
    )
  }

  if (isLoading) {
    return <LoadingIndicator />
  }

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Dividends' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />
        {renderContent()}
      </Grid>
    </Grid>
  )
}
