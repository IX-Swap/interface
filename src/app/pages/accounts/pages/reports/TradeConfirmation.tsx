import React from 'react'
import { Grid } from '@material-ui/core'
import { PageHeader } from 'app/components/PageHeader/PageHeader'
import { VSpacer } from 'components/VSpacer'
import { ReportsInfo } from 'app/pages/accounts/pages/reports/components/ReportsInfo/ReportsInfo'
import { ReportsAccordion } from 'app/pages/accounts/pages/reports/components/ReportsAccordion/ReportsAccordion'
import { useGetExchangeFills } from 'app/pages/accounts/hooks/useGetExchangeFills'

export const TradeConfirmation: React.FC = () => {
  const { data, isLoading } = useGetExchangeFills()
  console.log(data, isLoading)

  return (
    <Grid container direction={'column'}>
      <Grid item>
        <PageHeader title='Trade Confirmation' />
        <VSpacer size={'medium'} />
      </Grid>

      <ReportsInfo />

      <Grid item>
        <VSpacer size={'medium'} />

        <ReportsAccordion summary={'Trades'}>
          <div>Trades</div>
        </ReportsAccordion>
      </Grid>
    </Grid>
  )
}
