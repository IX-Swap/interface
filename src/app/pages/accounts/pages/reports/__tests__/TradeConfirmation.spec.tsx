import React from 'react'
import { render } from 'test-utils'
import { TradeConfirmation } from 'app/pages/accounts/pages/reports/TradeConfirmation'
import { fakeTradeItem } from '__fixtures__/reports'
import * as useTradeConfirmation from 'app/pages/accounts/hooks/useTradeConfirmation'
import { generateQueryResult } from '__fixtures__/useQuery'

describe('TradeConfirmation', () => {
  afterEach(async () => {})

  it('should match snapshot when data is undefined', () => {
    jest.spyOn(useTradeConfirmation, 'useTradeConfirmation').mockReturnValue(
      generateQueryResult({
        data: undefined,
        isLoading: false
      })
    )

    const { container } = render(<TradeConfirmation />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when data is loaded successfully', () => {
    jest.spyOn(useTradeConfirmation, 'useTradeConfirmation').mockReturnValue(
      generateQueryResult({
        data: [fakeTradeItem],
        isLoading: false
      })
    )

    const { container } = render(<TradeConfirmation />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot when data length is 0', () => {
    jest.spyOn(useTradeConfirmation, 'useTradeConfirmation').mockReturnValue(
      generateQueryResult({
        data: [],
        isLoading: false
      })
    )

    const { container } = render(<TradeConfirmation />)
    expect(container).toMatchSnapshot()
  })
})
