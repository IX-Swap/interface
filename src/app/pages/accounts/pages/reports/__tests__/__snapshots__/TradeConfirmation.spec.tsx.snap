// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeConfirmation should match snapshot when data is loaded successfully 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-container-14 css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container makeStyles-header-13 css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-i6prtc-MuiGrid-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-rldqdm-MuiTypography-root"
            >
              Trade Confirmation
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-128kl2r-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root css-12aw9rk"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-12aw9rk"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 makeStyles-root-19 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1m8eno4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded="false"
          class="MuiButtonBase-root makeStyles-summaryWrapper-20 MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="makeStyles-summaryContent-21 MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 makeStyles-summary-22 css-1okvd07-MuiTypography-root"
            >
              Trades
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-1cbajxi-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiAccordion-region"
                role="region"
              >
                <div
                  class="makeStyles-detailsWrapper-23 MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
                  >
                    <table
                      class="MuiTable-root css-7mlt34-MuiTable-root"
                    >
                      <thead
                        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-8dfetm-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1efwvfh-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          />
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1efwvfh-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          >
                            Date
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1efwvfh-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          >
                            Type
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1v6mra9-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          >
                            Quantity
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1v6mra9-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          >
                            Price
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-headColumn-24 css-1v6mra9-MuiTableCell-root"
                            data-testid="cell"
                            scope="col"
                          >
                            Total
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root makeStyles-row-29 makeStyles-row-34 css-8dfetm-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                            scope="row"
                          >
                            Securities Pair
                          </th>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          />
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          />
                        </tr>
                        <tr
                          class="MuiTableRow-root makeStyles-row-29 makeStyles-row-36 css-8dfetm-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                            scope="row"
                          >
                            HKUV/SGD
                          </th>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                          >
                            2021-05-28, 04:33 AM
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium makeStyles-column-26 css-17e0zod-MuiTableCell-root"
                          >
                            SELL
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          >
                            10
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          >
                            10.00
                          </td>
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium makeStyles-column-26 css-17jaypo-MuiTableCell-root"
                          >
                            100.00
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TradeConfirmation should match snapshot when data is undefined 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-container-3 css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container makeStyles-header-2 css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-i6prtc-MuiGrid-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-rldqdm-MuiTypography-root"
            >
              Trade Confirmation
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-128kl2r-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root css-12aw9rk"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-wrapper-8 css-l6hioc-MuiGrid-root"
      style="opacity: 0.75;"
    >
      <div
        class="MuiBox-root css-1jpm9pd"
      />
      <div
        class="MuiBox-root css-12aw9rk"
      />
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-q79gi7-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-12aw9rk"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-wrapper-9 css-t0zib5-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1kgakwq"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-12aw9rk"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 makeStyles-subtitle-10 css-1b4l2pg-MuiTypography-root"
          >
            This page is empty.
          </p>
          <div
            class="MuiBox-root css-7ieh0i"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 makeStyles-content-11 css-1b4l2pg-MuiTypography-root"
          >
            There are no records to display for the selected report.
          </p>
          <div
            class="MuiBox-root css-12aw9rk"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TradeConfirmation should match snapshot when data length is 0 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-container-40 css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container makeStyles-header-39 css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-i6prtc-MuiGrid-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-rldqdm-MuiTypography-root"
            >
              Trade Confirmation
            </h2>
          </div>
        </div>
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <nav
            class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-128kl2r-MuiTypography-root-MuiBreadcrumbs-root"
          >
            <ol
              class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
            />
          </nav>
        </div>
      </div>
      <div
        class="MuiBox-root css-12aw9rk"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-12aw9rk"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column makeStyles-wrapper-45 css-t0zib5-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1kgakwq"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-12aw9rk"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 makeStyles-subtitle-46 css-1b4l2pg-MuiTypography-root"
          >
            This page is empty.
          </p>
          <div
            class="MuiBox-root css-7ieh0i"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 makeStyles-content-47 css-1b4l2pg-MuiTypography-root"
          >
            There are no records to display for the selected report.
          </p>
          <div
            class="MuiBox-root css-12aw9rk"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
